onboardingtaskapp.client/src/components/Sale/Edit.jsx

```diff
@@ -19,44 +19,82 @@ export default function Edit({ item }) {
     const [stores, setStores] = useState([]);
 
     useEffect(() => {
-        fetch('https://localhost:7207/api/Customer')
-            .then(response => response.json())
-            .then(
-                data => setCustomers(data)
-            )
-            .catch(error => console.error('Error fetching customers:', error));
+        fetchCustomers()
+        fetchProducts()
+        fetchStores()
     }, []);
 
-    useEffect(() => {
-        fetch('https://localhost:7207/api/Product')
-            .then(response => response.json())
-            .then(data => setProducts(data))
-            .catch(error => console.error('Error fetching products:', error));
-    }, []);
+    
 
     useEffect(() => {
-        fetch('https://localhost:7207/api/Store')
-            .then(response => response.json())
-            .then(data => setStores(data))
-            .catch(error => console.error('Error fetching stores:', error));
-    }, []);
 
-    useEffect(() => {
-        const fetchSale = async () => {
-            try {
-                const response = await fetch(`https://localhost:7207/api/Sale/${item.sale.id}`);
-                if (!response.ok) {
-                    throw new Error('Network response was not ok');
-                }
-                const data = await response.json();
-                setFormData(data);
-            } catch (error) {
-                console.error('There was a problem fetching sale data:', error.message);
-            }
-        };
+        if (!item.sale.id) {
+            throw new Error('Invalid Id');
+        }
+               
         fetchSale();
+
     }, [item.sale.id]);
 
+    // const
+    const API_END_POINT = `https://onboardsite.azurewebsites.net/api/`;
```

Comment:
1. Please check my previous comment about console.logs
2. Please implement the API endpoint that I mentioned above